src/bindgen/parser.rs

```diff
@@ -156,6 +156,27 @@ impl<'a> Parser<'a> {
         assert!(self.lib.is_some());
         self.parsed_crates.insert(pkg.name.clone());
 
+        // Check if we should use cargo expand for this crate
+        if self.expand.contains(&pkg.name) {
+            return self.parse_expand_crate(pkg);
+        }
+
+        // Parse the crate before the dependencies otherwise the same-named idents we
+        // want to generate bindings for would be replaced by the ones provided
+        // by the first dependency containing it.
+        let crate_src = self.lib.as_ref().unwrap().find_crate_src(pkg);
+
+        match crate_src {
+            Some(crate_src) => self.parse_mod(pkg, crate_src.as_path())?,
+            None => {
+                // This should be an error, but is common enough to just elicit a warning
+                warn!(
+                    "Parsing crate `{}`: can't find lib.rs with `cargo metadata`.",
+                    pkg.name
+                );
+            }
+        }
+
         // Before we do anything with this crate, we must first parse all of its dependencies.
```

Comment:
What makes this not be needed then? If it's not really needed, then the comment should go away.

Reply:
good catch!